prow/plugins/pony/pony.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package pony adds pony images to issues in response to a /pony comment
+package pony
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+// Only the properties we actually use.
+type ponyResult struct {
+	Pony ponyResultPony `json:"pony"`
+}
+
+type ponyResultPony struct {
+	Representations ponyRepresentations `json:"representations"`
+}
+
+type ponyRepresentations struct {
+	Full  string `json:"full"`
+	Small string `json:"small"`
+}
+
+const (
+	ponyURL    = realHerd("https://theponyapi.com/pony.json")
```

Comment:
Is this the API with which we will need to be careful? Does it expose user-provided content? Do we run the risk of posting inappropriate images here?

Reply:
All the images returned by this API have been reviewed for appropriateness.

Reply:
@ixdy didn't we just have an oopsie with the cat plugin?

Reply:
@stevekuznetsov yes, there was apparently a bug in which images which were rejected were still available via a direct request. it's supposedly fixed now; all cat images are supposed to be reviewed first as well.